tests/phpunit/tests/general/wpPreloadLinks.php

```diff
@@ -0,0 +1,253 @@
+<?php
+
+/**
+ * @group general
+ * @group template
+ * @ticket 42438
+ * @covers ::wp_preload_resources
+ */
+class Tests_General_wpPreloadLinks extends WP_UnitTestCase {
+
+	/**
+	 * @dataProvider provider_preload_resources
+	 *
+	 * @ticket 42438
+	 */
+	public function test_preload_resources( $expected, $preload_resources ) {
+		$callback = function () use ( $preload_resources ) {
+			return $preload_resources;
+		};
+
+		add_filter( 'wp_preload_resources', $callback, 10 );
+		$actual = get_echo( 'wp_preload_resources' );
+		remove_filter( 'wp_preload_resources', $callback );
+
+		$this->assertSame( $expected, $actual );
+	}
+
+	/**
+	 * Test provider for all preload link possible combinations.
+	 *
+	 * @return array[]
+	 */
+	public function provider_preload_resources() {
```

Comment:
```suggestion
	public function data_preload_resources() {
```
Use `data` for the prefix per the docs.  (https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/)